tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -69,35 +75,100 @@ export const clickResource = async ({
 export interface createResourceArgs {
   page: Page
   name: string
-  type: 'folder'
+  type: 'folder' | 'txtFile' | 'mdFile' | 'drawioFile'
+  content?: string
 }
 
 export const createResource = async (args: createResourceArgs): Promise<void> => {
-  const { page, name } = args
+  const { page, name, type, content } = args
   const paths = name.split('/')
 
   for (const resource of paths) {
-    const folderExists = await resourceExists({
+    const resourcesExists = await resourceExists({
       page: page,
       name: resource
     })
 
-    if (!folderExists) {
+    if (!resourcesExists) {
       await page.locator(addNewResourceButton).click()
-      await page.locator(createNewFolderButton).click()
-      await page.locator(folderNameInput).fill(resource)
-      await Promise.all([
-        page.waitForResponse(
-          (resp) => resp.status() === 207 && resp.request().method() === 'PROPFIND'
-        ),
-        page.locator(actionConfirmationButton).click()
-      ])
-    }
 
-    await clickResource({ page, path: resource })
+      switch (type) {
+        case 'folder': {
+          await page.locator(createNewFolderButton).click()
+          await page.locator(resourceNameInput).fill(resource)
+          await Promise.all([
+            page.waitForResponse(
+              (resp) => resp.status() === 207 && resp.request().method() === 'PROPFIND'
+            ),
+            page.locator(actionConfirmationButton).click()
+          ])
+          break
+        }
+        case 'txtFile': {
+          await page.locator(createNewTxtFileButton).click()
+          await page.locator(resourceNameInput).fill(resource)
+          await Promise.all([
+            page.waitForResponse(
+              (resp) => resp.status() === 201 && resp.request().method() === 'PUT'
+            ),
+            page.locator(actionConfirmationButton).click()
+          ])
+          await editTextDocument({ page, content: content })
+          break
+        }
+        case 'mdFile': {
+          await page.locator(createNewMdFileButton).click()
+          await page.locator(resourceNameInput).fill(resource)
+          await Promise.all([
+            page.waitForResponse(
+              (resp) => resp.status() === 201 && resp.request().method() === 'PUT'
+            ),
+            page.locator(actionConfirmationButton).click()
+          ])
+          await editTextDocument({ page, content: content })
+          break
+        }
```

Comment:
How about minimizing duplication of code?

Reply:
here duplicated only 2 steps: `await page.locator(resourceNameInput).fill(resource)` and `page.locator(actionConfirmationButton).click()` with expecting response. to make a separate method for 2 steps? hm, I can but not sure

I did't move them to `editTextDocument()` because it only happens after the files are created